File: padroesEstruturais/padrao-de-projeto-bridge/php/src/racas/Humano.php
<?php

namespace Bridge\Php\racas;


use Bridge\Php\classes\Classe;

class Humano extends Raca
{
    const PODER_BASE = 1000.0;
    public function __construct(Classe $classe)
    {
        parent::__construct($classe);
    }


    public function calculaPoder(): float
    {
        return self::PODER_BASE * $this->getClasse()->getPoderBase();
    }

}

File: padroesEstruturais/padrao-de-projeto-bridge/java/src/main/java/racas/Humano.java
package racas;

import classes.Classe;

public class Humano extends Raca {

    private static final float PODER_BASE = 1000;
    public Humano(Classe classe) {
        super(classe);
    }

    @Override
    public float calculaPoder() {
        return PODER_BASE * this.getClasse().getPoderBase();
    }
}
